File: gestionApp/Empresa.java
package gestionApp;

import gestionApp.personas.Cliente;
import gestionApp.servicios.Servicio;

import java.util.ArrayList;

public class Empresa {

	private String id;
	private String nombre;
	private int nit;
	private String descripcion;

	private ArrayList<Cliente> clientes;
	private  ArrayList<Negocio> negociosAbiertos;
	private  ArrayList<Venta> ventas;
	private static ArrayList<Empresa> empresas;
	private  ArrayList<Servicio> serviciosDeInteres;

	private static int idCounter = 0;
    
	public Empresa(String nombre,int nit, String descripcion) {
		this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
		this.nombre = nombre;
		this.nit = nit;
		this.descripcion = descripcion;
		this.negociosAbiertos = new ArrayList<>();
		this.ventas = new ArrayList<>();
		this.serviciosDeInteres = new ArrayList<>();
		idCounter += 1;
	}

	// getters y setters 
	public String getId(){return this.id;}

	public String getNombre() {return nombre;}
	public void setNombre(String nombre) {this.nombre = nombre;}

	public int getNit() {return nit;}
	public void setNit(int nit) {this.nit = nit;}

	public String getDescripcion() {return descripcion;}
	public void setDescripcion(String descripcion) {this.descripcion = descripcion;}

	public ArrayList<Cliente> getClientes() {return clientes;}
	public void setClientes(Cliente nuevoClientes) {this.clientes.add(nuevoClientes);}

	public ArrayList<Negocio> getNegociosAbiertos() {return negociosAbiertos;}
	public void setNegociosAbiertos(Negocio nuevoContacto) {this.negociosAbiertos.add(nuevoContacto);}

	public ArrayList<Venta> getVentas() {return ventas;}
	public void setVenta(Venta nuevaVenta) {this.ventas.add(nuevaVenta);}

	public static ArrayList<Empresa> getEmpresas() {return empresas;}
	public static void setEmpresas(Empresa nuevaEmpresa) {Empresa.empresas.add(nuevaEmpresa);}

	@Override
	public  String toString(){
		return "\tDatos del Empresa \nEmpresa: "+ nombre +
		 "\nNit de la empresa: " + nit + 
		 "\nID: " + id +
		 "\nDescripcion de la empresa: " + descripcion +
		 "\nNegocios Abiertos: " + negociosAbiertos +
		 "\nVentas: " + ventas +
		 "\nServicios de Interes: " + serviciosDeInteres;

		
	}

	

	//FUNCIONALIDADES

}

File: gestionApp/Empresa.py
from tkinter import messagebox
from gestionApp.excepciones import *

class Empresa:
    
    allEmpresas = []
    clientes = []
    negociosAbiertos = []
    ventas = []
    serviciosInteres = []
    def __init__(self, nombre, nit, description):
        self.nombre = nombre
        self.nit = nit
        self.description = description
        self.allEmpresas.append(self)
        messagebox.showinfo(message ='EMPRESA creada correctamente', title = 'Creacion Empresa')

    def getNombre(self): return self.nombre
    def setNombre(self,nombre): self.nombre = nombre
    
    def getNit(self): return self.nombre
    def setNit(self,nit): self.nit = nit
    
    def getDescripcion(self): return self.description
    def setDescripcion(self,description): self.description = description
        
    @classmethod    
    def getAllEmpresas(cls): return cls.allEmpresas
    @classmethod
    def setAllEmpresas(cls,allEmpresas): cls.allEmpresas = allEmpresas
    
    @classmethod
    def getClientes(cls): return cls.clientes
    @classmethod
    def setClientes(cls,clientes): cls.clientes = clientes
    
    @classmethod
    def getNegociosAbiertos(cls): return cls.negociosAbiertos
    @classmethod
    def setNegociosAbiertos(cls,negociosAbiertos): cls.negociosAbiertos = negociosAbiertos
    
    @classmethod
    def getVentas(cls): return cls.ventas
    @classmethod
    def setVentas(cls,ventas): cls.ventas = ventas 
    
    @classmethod
    def getServiciosInteres(cls): return cls.serviciosInteres
    @classmethod
    def setServiciosInteres(cls,serviciosInteres): cls.serviciosInteres = serviciosInteres

    @classmethod
    def busquedaEmpresa(cls, nit):
        for i in range(len(Empresa.getAllEmpresas())):
            if nit == str:
                try:
                    raise ExcepcionCamposNulos()
                except ExcepcionCamposNulos as f:
                    f.showMessage()
                return
            if Empresa.getAllEmpresas()[i].nit == nit:
                return Empresa.getAllEmpresas()[i]

    @classmethod
    def eliminarEmpresa(cls, nit):
        for i in range(len(Empresa.getAllEmpresas())):
            if nit == str:
                try:
                    raise ExcepcionCamposNulos()
                except ExcepcionCamposNulos as f:
                    f.showMessage()
                return           
            if Empresa.allEmpresas[i].nit == nit:
                Empresa.getAllEmpresas().pop(i)
        messagebox.showinfo(message='EMPRESA eliminada correctamente', title='Eliminacion Empresa')

    def infoEmpresa(self):
        return f'\t---Informacion Empresa---  \n Nombre: {self.nombre} \n Nit: {self.nit} \n Descripcion: {self.descripcion}'

                   






